### ISSUE TITLE
--no-bin-links options skip the .bin folder creation

### ISSUE THREAD

**Issue by fhewitt** [yarnpkg/yarn#4782]
**Do you want to request a *feature* or report a *bug*?**
*bug*

**What is the current behavior?**
`yarn install --no-bin-links` skip the creation of the node_modules/.bin folder

This broke the installation and prevent usage of CLI command in yarn scripts. Our current workaround is to install globally the dependencies for which we need the CLI commands, but this is a wrong solution.

**What is the expected behavior?**
.bin folder should be created, with copied files. This is the behaviour on Window without the --no-bin-links flag.

This is essential to be able to use yarn inside a VM, where you need --no-bin-links, but don't want to lost functionalities.

**Please mention your node.js, yarn and operating system version.**
node 8.6.0
yarn 1.2.1.
Window 10 / Ubuntu 14.04 through Vagrant on Window 10.

**Comment by fdelprete** [comment]
Same behavior in my environment:

- node v6.11.3
- yarn 1.3.2
- CentOs 7.4.1708 (Core) through Vagrant on Windows 10

**Comment by W1M0R** [comment]
I have the same issue. I am using Docker Toolbox on Windows 10, sharing the node_modules on the Docker host (Windows) as a volume in the Docker container, to facilitate caching.

**Comment by smxsm** [comment]
Yep, that's a problem - "--no-bin-links" is needed on Windows inside e.g. a Vagrant VM since Symlinks don't really work there

**Comment by trompx** [comment]
Did someone came up with a better solution than installing modules globally?
I tried to use their full path directly in yarn scripts but then I get permission errors like:
```
$ yarn ./node_modules/lerna bootstrap
/bin/sh: ./node_modules/lerna: Permission denied
```
even setting node_modules permissions to 777

**Comment by trompx** [comment]
Ok for those looking for an alternative solution, I made it work.
You must run your command with **node** prefix.
`$ yarn node ./node_modules/lerna bootstrap`

**Comment by waclock** [comment]
The problem arises when you have postinstall scripts in certain libraries that make use of these symlinks and/or the .bin folder.
